pkg/diskmaker/controllers/deleter/reconcile.go

```diff
@@ -0,0 +1,102 @@
+	// "sigs.k8s.io/controller-runtime/pkg/reconcile"
+	// provCommon "sigs.k8s.io/sig-storage-local-static-provisioner/pkg/common"
+	staticProvisioner "sigs.k8s.io/sig-storage-local-static-provisioner/pkg/common"
+)
+
+// Reconcile reads that state of the cluster for a LocalVolumeSet object and makes changes based on the state read
+// and what is in the LocalVolumeSet.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileDeleter) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling LocalVolumeSet")
+	// enqueue if cache is not initialized
+	// and if any pv has phase == Releaseds
+
+	// mux for runtimeConfig access
+	r.provisionerMux.Lock()
+	defer r.provisionerMux.Unlock()
+
+	// get associated provisioner config
+	cm := &corev1.ConfigMap{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: common.ProvisionerConfigMapName, Namespace: request.Namespace}, cm)
+	if err != nil {
+		reqLogger.Error(err, "could not get provisioner configmap")
+		return reconcile.Result{}, err
+	}
+
+	// read provisioner config
+	provisionerConfig := staticProvisioner.ProvisionerConfiguration{}
+	staticProvisioner.ConfigMapDataToVolumeConfig(cm.Data, &provisionerConfig)
+
+	r.runtimeConfig.DiscoveryMap = provisionerConfig.StorageClassConfig
+	r.runtimeConfig.NodeLabelsForPV = provisionerConfig.NodeLabelsForPV
+	r.runtimeConfig.Namespace = request.Namespace
+	r.runtimeConfig.SetPVOwnerRef = provisionerConfig.SetPVOwnerRef
+	r.runtimeConfig.Name = common.GetProvisionedByValue(*r.runtimeConfig.Node)
+
+	// ignored by our implementation of static-provisioner,
+	// but not by deleter (if applicable)
+	r.runtimeConfig.UseNodeNameOnly = provisionerConfig.UseNodeNameOnly
+	r.runtimeConfig.MinResyncPeriod = provisionerConfig.MinResyncPeriod
+	r.runtimeConfig.UseAlphaAPI = provisionerConfig.UseAlphaAPI
+	r.runtimeConfig.LabelsForPV = provisionerConfig.LabelsForPV
+
+	// initialize the pv cache
+	// initialize the deleter's pv cache on the first run
+	if !r.firstRunOver {
+		log.Info("initializing PV cache")
+		pvList := &corev1.PersistentVolumeList{}
+		err := r.client.List(context.TODO(), pvList)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to initialize PV cache")
+		}
+		for _, pv := range pvList.Items {
+			// skip non-owned PVs
+			name, found := pv.Annotations[provCommon.AnnProvisionedBy]
+			if !found || name != r.runtimeConfig.Name {
+				continue
+			}
+			addOrUpdatePV(r.runtimeConfig, pv)
+		}
+
+		r.firstRunOver = true
+	}
+
+	// get the lock
+	// run the deleter
+	//      for each pv with phase == released, mark cleanup running and run cleanup.
+	r.deleter.DeletePVs()
+	return reconcile.Result{}, nil
```

Comment:
Since this code adds new labels to PVs:

```
    storage.openshift.com/device-id: dm-name-sandbox-rails
    storage.openshift.com/device-name: dm-0
    storage.openshift.com/owner-kind: LocalVolume
    storage.openshift.com/owner-name: example
    storage.openshift.com/owner-namespace: default
```

In particular I think there were some known bugs around device-id and device-name - https://bugzilla.redhat.com/show_bug.cgi?id=1933630 . This would result in regression of `LocalVolume` functionality, which is GA.

Reply:
will move both to annotations

Reply:
Done